### PR TITLE
Popover test

### PR DESCRIPTION
**Pull request by ajrothwell**
this is to bring into master all changes made over the past week that adapts layerboard to use the Popover.vue and PopoverLink.vue components of pvc

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,9 +11,9 @@
     "build": "vue-cli-service build"
   },
   "dependencies": {
-    "@philly/vue-comps": "^1.0.28",
-    "@philly/vue-datafetch": "^0.0.16",
-    "@philly/vue-mapping": "^1.0.29",
+    "@philly/vue-comps": "https://github.com/CityOfPhiladelphia/phila-vue-comps#c381269ee0aa5e378dac374e1a8997993a25009f",
+    "@philly/vue-datafetch": "https://github.com/CityOfPhiladelphia/phila-vue-datafetch#b2a06c17727bddf5a59b75bab6e4965a24f66484",
+    "@philly/vue-mapping": "https://github.com/CityOfPhiladelphia/phila-vue-mapping#262bfebec8f567ee6df96fd3fa8fe18ccf4f016a",
```
Comment (danlopez) [inline package.json:16]: @ajrothwell - any reason why we aren't publishing the versions of these libraries that we're using? Are they diverging from the published versions?

File: src/components/Layerboard.vue [hunk 2]
```diff
@@ -88,6 +109,10 @@
         }
       }
 
+      if (this.$config.dataSources) {
+        this.$controller.dataManager.fetchData();
```
Comment (danlopez) [inline src/components/Layerboard.vue:113]: Is this operation async? If so, I wonder if we should be using async... await to ensure that we're not seeing race conditions later in the code.

File: src/components/LbFooter.vue [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+              return this.$config.footer.helpPopover.height
+            } else {
+              return '80%'
+            }
+          } else {
+            return '80%'
+          }
+        } else {
+          return '80%'
+        }
```
Comment (danlopez) [inline src/components/LbFooter.vue:73]: I feel like there's a cleaner way to write these conditionals... nested if statements can be _so_ hard to read.

Comment (danlopez) [reply src/components/LbFooter.vue:73]: It looks like we could rewrite this as: 
```
popoverHeight() {
if (this.$config.footer & this.$config.footer.helpPopover && this.$config.helpPopover.height) {
  return this.$config.footer.helpPopover.height;
}

return 80%;
```
}

File: src/components/MapPanel.vue [hunk 4]
```diff
@@ -368,6 +368,13 @@
           return null
         }
       },
+      basemapSelectControlPosition() {
+        if (this.isMobileOrTablet) {
+          return 'topright'
+        } else {
+          return 'topalmostright'
```
Comment (danlopez) [inline src/components/MapPanel.vue:375]: Note that we don't need an else if we have a `return` in the other branch of the conditional.

File: src/components/MapPanel.vue [hunk 5]
```diff
@@ -425,7 +432,11 @@
             topicLayers = component.options.topicLayers;
           }
         }
-        return topicLayers;
+        let topicLayersKeys = [];
+        for (let topicLayer of topicLayers) {
+          topicLayersKeys.push(topicLayer.title)
+        }
```
Comment (danlopez) [inline src/components/MapPanel.vue:438]: I think we could rewrite this (lines 435 - 438) as something like: 
```
topicLayerKeys = topicLayers.map(topicLayer =>  topicLayer.title);
```

**Review by danlopez — COMMENTED** [review]
Left some post-merge review comments here.

File: package.json (hunk 1 above)
Comment (ajrothwell) [reply package.json:16]: I don't want to do actual npm releases that frequently

File: src/components/Layerboard.vue (hunk 2 above)
Comment (ajrothwell) [reply src/components/Layerboard.vue:113]: yes, fetchData() is an enormous function that loops ajax calls. Happy to look at it with you and try to use promises to make it more reliable and efficient some time

File: src/components/MapPanel.vue (hunk 5 above)
Comment (ajrothwell) [reply src/components/MapPanel.vue:438]: thanks.  Similarly, I'm actually looking up .filter() and reminding myself how it works right now for the work I'm doing to make the checkbox ordering work.  I knew about .map(), but have never made it my habit to use it yet

**Pull request merged**